### ISSUE TITLE
Faulty compiled SVG path?

### ISSUE THREAD

**Issue by newtonianb** [tcoopman/image-webpack-loader#14]
I installed this plugin then added this to my webpack plugins

**webpack.config**
```
                {
                    test: /\.(jpe?g|png|gif|svg)$/i,
                    loaders: [
                        'file?hash=sha512&digest=hex&name=[hash].[ext]',
                        'image-webpack?bypassOnDebug&optimizationLevel=7&interlaced=false'
                    ]
                },
```

now I have one entry point **app.js**
```
require('styles/icons')
```

**styles/icons.scss** whose content is
```
@each $keyword in 'fn', 'dn' {
  @each $color-keyword in 'red', 'white' {

    .icon--#{$keyword}--#{$color-keyword} {
      background: url("../svg/icon--#{$keyword}--#{$color-keyword}.svg") no-repeat left top;
      background-size: contain;
    }

    .svg-off .icon--#{$keyword}--#{$color-keyword} {
      background-image: url("../images/icon--#{$keyword}--#{$color-keyword}.png");
    }
  }
}
```

This compiles fine and when I load my webpage I see 
```
.icon--fn--red {
  background: url(/assets/6e4ae90369b7deca283f326cdb52060e.svg) no-repeat left top;
  background-size: contain;
}
```

But then I look at that file **/assets/6e4ae90369b7deca283f326cdb52060e.svg** and I see 
```
module.exports = __webpack_public_path__ + "695fb5da0124fe2dd0c33bab5f22f59e.svg"
```

I checked the file **695fb5da0124fe2dd0c33bab5f22f59e.svg** and it points to the correct svg.

When I try to access **/assets/6e4ae90369b7deca283f326cdb52060e.svg** in the browser I get
```
This page contains the following errors:

error on line 1 at column 1: Document is empty
Below is a rendering of the page up to the first error.
```

How do I fix this?

**Comment by tcoopman** [comment]
Without your full webpack config this is a bit hard to say.
Do you have the same problem if you remove the `file-loader`?

If the svg file itself is correct I would guess this is a problem with `file-loader` or your webpack config.

The `image-loader` only optimizes the svg, the filename is decided by the `file-loader` and the assets path by webpack.

**Comment by newtonianb** [comment]
Hi @tcoopman, I just realised I also had this configuration ``{test: /\.svg(\?v=\d+\.\d+\.\d+)?$/, loader: "file"},`` when I removed it I now get this on compilation

```
Module parse failed:
D:\prj\a\node_modules\font-awesome\fonts\fontawesome-webfont.svg?v=4.3.0 Line 1:
Unexpected token <
You may need an appropriate loader to handle this file type.
| <?xml version="1.0" standalone="no"?>
| <!DOCTYPE svg PUBLIC "-//W3C//DTD SVG 1.1//EN" "http://www.w3.org/Graphics/SVG/1.1/DTD/svg11.dtd" >
| <svg xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" version="1.1">
```


My full webpack config is the following

```
var webpackConfig = {
        context: jsSrc,

        resolve: {
            extensions: ['', '.js', '.coffee', '.scss', '.css'],
            modulesDirectories: [node_dir, bower_dir]
        },

        module: {
            loaders: [
                {
                    test: /[\\\/]bower_components[\\\/]modernizr[\\\/]modernizr\.js$/,
                    loader: "imports?this=>window!exports?window.Modernizr"
                },
                {
                    test: /\.(jpe?g|png|gif|svg)$/i,
                    loaders: [
                        'file?hash=sha512&digest=hex&name=[hash].[ext]',
                        'image-webpack?bypassOnDebug&optimizationLevel=7&interlaced=false'
                    ]
                },
                {
                    test: /\.coffee$/, loader: 'coffee-loader'
                },                
                {
                    test: /\.woff(\?v=\d+\.\d+\.\d+)?$/, loader: "file"
                },
                {
                    test: /\.woff2(\?v=\d+\.\d+\.\d+)?$/, loader: "file"
                },
                {
                    test: /\.ttf(\?v=\d+\.\d+\.\d+)?$/, loader: "file"
                },
                {
                    test: /\.eot(\?v=\d+\.\d+\.\d+)?$/, loader: "file"
                },
                {
                    test: /\.svg(\?v=\d+\.\d+\.\d+)?$/, loader: "file"
                },
                {
                    test: /\.scss$/,
                    loader: 'style-loader!css-loader!sass-loader?' + JSON.stringify({ includePaths: [ stylesSrc ], sourceMap : env == 'development' })
                },
                {
                    test: /\.css$/,
                    loader: "style-loader!css-loader"
                }
            ],
            noParse: [],
        },

        entry: {
            app: ['./app.coffee']
        },

        output = {
            path: jsDest,
            filename: '[name].js',
            publicPath: '/assets/',
            chunkFilename: "/js/[id].js"
        }        
    }
```

**Comment by tcoopman** [comment]
If you want the _font-awesome\fonts\fontawesome-webfont.svg?v=4.3.0_ to be run through the image-loader you should change the `test: /\.svg(\?v=\d+\.\d+\.\d+)?$/,` line to include the image-loader.

If you remove that test, the regex `/\.(jpe?g|png|gif|svg)$/i` won't match _webfont.svg?v=4.3.0_, that's why you get the other error when you remove that line. That also means that your initial issue has nothing to do with image-loader as the file is not run through the image-loader at all, so I'm going to close this.

I would think that your problem has something to do with how you run/access your code. How do you do that? With webpack-dev-server?

I also see that your image-loader doesn't include svg optimization at the moment, if you want those, you need to add some `svgo` options. See https://github.com/tcoopman/image-webpack-loader/blob/master/test/webpack.config.js for an example.

**Comment by newtonianb** [comment]
Ah yes I didn't realize font-awesome css had this added get parameter to the font path thank you.
I am currently using gulp with browsersync, webpack calls browsersync to reload on watch 

```
    webpack(config).watch(200, function (err, stats) {
        logger(err, stats)
        browserSync.reload()
        // On the initial compile, let gulp know the task is done
        if (!built) {
            built = true;
            callback()
        }
    })
```

**Comment by tcoopman** [comment]
Is the `jsDest` folder correct?
I haven't used browserSync so I'm not sure that is the problem or not, maybe you can look further in that direction. Maybe try to reproduce it with an image file in your js files?

**Comment by newtonianb** [comment]
Yes that all works fine (its a variable defined above). It all works great except with that SVG loader issue. The images loading have been working with the image loader, I think the main issue was what you pointed out, the regex doesn't account for the version at the end.

The issue became clear when I remove the original faulty SVG loader that overwrote image-webpack-loader. Once I only used the image-webpack-loader I started getting useful errors.
